sdk/nodejs/tests/runtime/langhost/cases/075.invoke_depends_on/index.js

```diff
@@ -0,0 +1,24 @@
+// Test the dependsOn invoke option.
+
+const assert = require("assert");
+const pulumi = require("../../../../../");
+
+const dependency = { resolved: false };
+
+const dependsOn = pulumi.output(
+    new Promise((resolve) =>
+        setTimeout(() => {
+            dependency.resolved = true;
+            resolve();
+        }),
+    ),
+);
+
+// By the time we serialise the arguments of the invoke, dependency.resolved
+// should be true due to dependsOn awaiting the setTimeout promise resolution
+// above. Without dependsOn, the invoke will be serialised before promise
+// resolution (since promises [microtasks] happen before timeout events
+// [macrotasks]).
+pulumi.runtime.invoke("test:index:echo", { dependency }, { dependsOn }).then((result) => {
```

Comment:
Does this type check? Doesn't dependsOn need to return a Resource? We'd just mock it out anyway, but curious as to what's letting this work as is.

Reply:
PlEaSiNgLy the tests are written in untyped JavaScript, so this is an abuse. I cribbed this from `depends_on_non_resource`, which I realise now is designed to catch a deliberate error. I can rework -- if so I _guess_ the `setTimeout` hackery would move to the spec-side mock of `RegisterResource`? I wonder if that works... checking

Reply:
I think the fix is simpler actually -- just resolve to a `Resource`. I've done this and it seems happy (and in theory we are testing an actual scenario now). I've also added a test for non-resource depends, which it seems happy with too.